hack/tools/janitor/janitor_test.go

```diff
@@ -0,0 +1,566 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/simulator"
+	"github.com/vmware/govmomi/simulator/vpx"
+	"github.com/vmware/govmomi/view"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/internal/test/helpers/vcsim"
+)
+
+func setup(ctx context.Context, t *testing.T) (*vSphereClients, *vcsim.Simulator) {
+	t.Helper()
+	model := &simulator.Model{
+		ServiceContent: vpx.ServiceContent,
+		RootFolder:     vpx.RootFolder,
+		Autostart:      true,
+		Datacenter:     1,
+		Portgroup:      1,
+		Host:           1,
+		Cluster:        1,
+		ClusterHost:    3,
+		DelayConfig:    simulator.DelayConfig{},
+	}
+
+	vcsim, err := vcsim.NewBuilder().WithModel(model).Build()
+	if err != nil {
+		panic(fmt.Sprintf("unable to create simulator %s", err))
+	}
+
+	fmt.Printf(" export GOVC_URL=%s\n", vcsim.ServerURL())
+	fmt.Printf(" export GOVC_USERNAME=%s\n", vcsim.Username())
+	fmt.Printf(" export GOVC_PASSWORD=%s\n", vcsim.Password())
+	fmt.Printf(" export GOVC_INSECURE=true\n")
+
+	clients, err := newVSphereClients(ctx, getVSphereClientInput{
+		Username:  vcsim.Username(),
+		Password:  vcsim.Password(),
+		Server:    vcsim.ServerURL().String(),
+		UserAgent: "capv-janitor-test",
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	t.Cleanup(vcsim.Destroy)
+
+	return clients, vcsim
+}
+
+func setupTestCase(g *gomega.WithT, sim *vcsim.Simulator, objects []*simCreator) (string, map[string]bool) {
+	g.THelper()
+
+	relativePath := rand.String(10)
+
+	baseRP := vcsimResourcePool("")
+	baseFolder := vcsimFolder("")
+	baseDatastore := vcsimDatastore("", os.TempDir())
+	// Create base objects for the test case
+	g.Expect(baseRP.Create(sim, relativePath)).To(gomega.Succeed())
+	g.Expect(baseFolder.Create(sim, relativePath)).To(gomega.Succeed())
+	g.Expect(baseDatastore.Create(sim, relativePath)).To(gomega.Succeed())
+
+	createdObjects := map[string]bool{}
+
+	// Create objects
+	for _, object := range objects {
+		splitted := strings.Split(object.Path(relativePath), relativePath+"/")
```

Comment:
Why is relativePath/ the separator? How often does this occur in object.Path(relativePath)?

Reply:
Or is this just a prefix or suffix?

Reply:
Isn't splitted[1] always c.p?